server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -438,6 +439,34 @@ public Map<String, NamedAnalyzer> indexAnalyzers() {
         return Map.of();
     }
 
+    /**
+     * If a mapper natively supports synthetic source, f.e. by constructing it from doc values.
+     * @return true or false
+     */
+    protected boolean supportsSyntheticSourceNatively() {
+        return false;
+    }
+
+    /**
+     * If a mapper opts in to use fallback synthetic source implementation.
+     * @return true or false
+     */
+    protected boolean fallbackSyntheticSourceOptIn() {
```

Comment:
If implementations overwrite this method, why would these implementations implement synthetic source natively? It is the same as `supportsSyntheticSourceNatively()` returning false? 

Maybe we can just have `supportsSyntheticSourceNatively()`?

Reply:
This is a temporary flag just to be able to introduce support on a per-field basis instead of one big PR. We will remove this when we enable synthetic source support for the last field.